modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -926,6 +959,23 @@ public CompletableFuture<Void> stopAsync(ComponentContext componentContext) {
         shutdownAndAwaitTermination(executor, shutdownTimeoutSeconds, TimeUnit.SECONDS);
         shutdownAndAwaitTermination(replicasCreationExecutor, shutdownTimeoutSeconds, TimeUnit.SECONDS);
 
+        try {
+            CompletableFuture.allOf(raftClientsFutures.values().toArray(new CompletableFuture<?>[0]))
+                    .get(shutdownTimeoutSeconds, TimeUnit.SECONDS);
+
+            raftClientsFutures.forEach((raftNodeId, raftClientFuture) -> {
+                raftClientFuture.thenAccept(RaftGroupService::shutdown);
+
+                try {
+                    raftManager.stopRaftNode(raftNodeId);
+                } catch (NodeStoppingException e) {
+                    LOG.debug("Raft node is already stopping [raftNodeId={}]", raftNodeId, e);
+                }
+            });
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
```

Comment:
Please use graceful stop, like org.apache.ignite.internal.util.IgniteUtils#closeAll(java.util.stream.Stream<? extends java.lang.AutoCloseable>)

Reply:
Raft client shutdown process is removed now in favor of https://issues.apache.org/jira/browse/IGNITE-23109